lib/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -1,72 +1,98 @@
 import { useId } from 'react';
-import classNames from 'classnames';
 
-export interface DropdownMenuOptionsData<T extends string = string> {
-  value: T;
+export interface DropdownMenuSelectProps<T extends DropdownMenuOptionsData> {
+  id: string;
+  disabled?: boolean;
+  ariaLabel?: string;
+  hideLabel?: boolean;
+  options: T[];
+  defaultValue?: T;
+  selected?: T;
+  onChange?: (value: T) => void;
+}
+
+const DropdownMenuSelect = <T extends DropdownMenuOptionsData = DropdownMenuOptionsData>({
+  id,
+  disabled,
+  ariaLabel,
+  onChange,
+  defaultValue,
+  options,
+}: DropdownMenuSelectProps<T>) => (
+  <select
+    id={id}
+    disabled={disabled}
+    aria-label={ariaLabel}
+    className="min-w-[120px] justify-self-end rounded-lg border border-jod-dark bg-jod-white p-2 hover:bg-purple-100 focus:outline-none focus:ring focus:ring-purple-500 disabled:border-gray-500 disabled:bg-gray-200 disabled:text-gray-500 disabled:hover:bg-gray-200"
+    onChange={(event: React.ChangeEvent<HTMLSelectElement>) => {
+      if (onChange) {
+        onChange(options.find((option) => option.value === event.target.value)!);
+      }
+    }}
+    defaultValue={defaultValue?.value}
+  >
+    {options.map((option, index) => (
+      <option key={index} value={option.value}>
+        {option.label}
+      </option>
+    ))}
+  </select>
+);
+
+export interface DropdownMenuOptionsData {
+  value: string;
   label: string;
 }
 
-interface DropdownMenuProps<T extends DropdownMenuOptionsData, U extends string = string> {
+interface DropdownMenuProps<T extends DropdownMenuOptionsData> {
```

Comment:
Rationale for the old use of generics was that it gives more type safety if needed to ensure that only correct values can be given for example.
Previously you could allow e.g only certain languages:
```tsx
type LanguageCode = 'fi' | 'sv' | 'en';
// ..
// FAIL: "ee" is not one of the allowed values here
render(<DropdownMenu<LanguageCode> label="" options={options} defaultValue="ee" />);

// SUCCESS: "en" is an allowed value
render(<DropdownMenu<LanguageCode> label="" options={options} defaultValue="en" />);
```

And it can also be sure that only correct **Language** in this case can be returned from the `onChange`.
```tsx
render(<DropdownMenu<Language> label="" options={options} defaultValue="sv" onChange={(newLang: Language) => console.log('TODO: some real logic based on the language', newLang)} />);
```

Original PR: https://github.com/Opetushallitus/jod-yksilo-ui/pull/23


But maybe generics are too complicated for this library to be used and these type errors can be prevented earlier, so the actual simplification of this PR is welcome.

Reply:
Hmm, I think default value should be one value of `options`. For example, let's say that `options` are:
```typescript
const options = [
  {
    value: 'apple'
    label: 'Apple'
  },
  {
    value: 'banana'
    label: 'Banana'
  }
]
```
In this case, the default value should be only `apple` or `banana` (or undefined). I'm not sure if such typings are even possible.

Reply:
Original thought was the label to be just translation, visual for the user. And the value is the actual interesting part.

But of course if the actual **option** item has additional data added by the user, it might be smarter to have the default value to be object instead.

Reply:
defaultOption is a valid React prop for select. Therefore, the prop name is also defaultOption.

Reply:
I undo these typings changes.